### DISCUSSION TITLE
Question mark in directory/file names changed to underscore?

### DISCUSSION THREAD

**Discussion by er013** [transmission/transmission#2456] [Q&A] [answered]
Discovered one torrent just now, among few hundred ones I keep sharing, that says no data found and suggests to remove and re-add torrent to fix tha situation. It has been in my transmission-gtk for some time, share ratio is 85 and data certainly is present as it should be. And its read-only share so no way transmission or any other mechanism would change said data... Just transmission instance for long-term sharing downloaded torrents.

Tried removing and re-adding. No dice, still transmission feels liks there is no pre-existing data present. Tried starting download to writable directory and this revealed that directory name and also filename inside said directory is different now. Originally both contain questionmark. And downloading that same torrent now end up directory and file names containing underscore instead. 

Torrent hash in question, if you want to try it yourself: 2f2a2d80744af1376340a847e4dea9ff1cf034c6

Dont know when such change happened, currently running version is 3.00-1 on debian.

So, how come questionmark in dir/file names is suddenly prohibited? Any plans for reverting such a change?

**Comment by ckerr** [comment, marked as answer]
Creating torrents with a questionmark in the filename is bad form, since that is disallowed on some platforms. See https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions for more information.

No plans to revert the convert-to-underscore behavior, since undoing that would make those torrents unusable on those platforms.
